include/libstatistics_collector/topic_statistics_collector/received_message_age.hpp

```diff
@@ -103,6 +114,8 @@ class ReceivedMessageAgeCollector : public TopicStatisticsCollector<T>
   * @param received_message the message to calculate age of.
   * @param now_nanoseconds time the message was received in nanoseconds
   */
+  [[deprecated("Don't use ReceivedMessageAgeCollector with type T, use rmw_message_info_t"
```

Comment:
deprecation period is given to only this method? probably we want to deprecate `template<typename T> ReceivedMessageAgeCollector` class, and remove on turtle-K?

Reply:
https://github.com/ros-tooling/libstatistics_collector/pull/170#issuecomment-1718419858 explains the weird behavior that comes with deprecating a templated class. While it would make the most sense to deprecate that class, I had weird troubles getting rid of it.